x-pack/plugins/fleet/public/applications/fleet/sections/debug/components/integration_debugger.tsx

```diff
@@ -0,0 +1,242 @@
+      </EuiTitle>
+
+      <EuiSpacer size="m" />
+
+      <EuiText grow={false}>
+        <p>Use this tool to uninstall or reinstall installed integrations.</p>
+        <p>
+          Reinstalling an integration will uninstall and then immediately install it again. This can
+          be useful in restoring broken or malformed integration installations when a user hasn{"'"}
+          t done much customization.
+        </p>
+      </EuiText>
+
+      <EuiSpacer size="m" />
+
+      <EuiFlexGroup alignItems="center" justifyContent="flexStart">
+        <EuiFlexItem
+          grow={false}
+          css={`
+            min-width: 600px;
+          `}
+        >
+          <EuiComboBox
+            aria-label="Select an Integration"
+            placeholder="Select an Integration"
+            fullWidth
+            options={comboBoxOptions}
+            singleSelection={{ asPlainText: true }}
+            selectedOptions={selectedOptions}
+            isDisabled={
+              integrations.status === 'loading' ||
+              reinstallMutation.isLoading ||
+              uninstallMutation.isLoading
+            }
+            prepend={
+              selectedOptions.length > 0 ? (
+                <EuiButtonEmpty>
+                  <EuiIcon type={selectedOptions[0]?.icon ?? 'fleetApp'} />
+                </EuiButtonEmpty>
+              ) : undefined
+            }
+            renderOption={(option, searchValue, contentClassName) => (
+              <span className={contentClassName}>
+                <EuiIcon type={(option as any).icon} />
+                &nbsp;
+                <EuiHighlight search={searchValue}>{option.label}</EuiHighlight>
+              </span>
+            )}
+            onChange={(newSelectedOptions) => {
+              // Handle "clear" action
+              if (!newSelectedOptions.length) {
+                setSelectedIntegrationId(undefined);
+              } else {
+                setSelectedIntegrationId(newSelectedOptions[0].value);
+              }
+            }}
+          />
+        </EuiFlexItem>
+
+        {selectedIntegration && (
+          <EuiFlexGroup>
+            <EuiFlexItem grow={false}>
+              <EuiButton
+                color="primary"
+                isLoading={reinstallMutation.isLoading}
+                onClick={() => setIsReinstallModalVisible(true)}
+              >
+                Reinstall
+              </EuiButton>
+            </EuiFlexItem>
+
+            <EuiFlexItem grow={false}>
+              <EuiButton
+                color="danger"
+                isLoading={uninstallMutation.isLoading}
+                onClick={() => setIsUninstallModalVisible(true)}
+              >
+                Uninstall
+              </EuiButton>
+            </EuiFlexItem>
+
+            {isReinstallModalVisible && (
+              <EuiConfirmModal
+                title={`Reinstall ${selectedIntegration.title}`}
+                onCancel={() => setIsReinstallModalVisible(false)}
```

Comment:
I think there is a small bug when clicking confirm, `setIsReinstallModalVisible(false)` is not called, and the modal becomes visible again when selecting another integration.

Reply:
Yep good call! This is fixed now. Going to merge once CI is green. Thanks!